internal/impl/aws/helper/iter.go

```diff
@@ -0,0 +1,44 @@
+package helper
+
+import (
+	"iter"
+)
+
+// TokenIterator creates an iterator from a token-based pagination function `fn`.
+// Iteration stops when nextToken is nil or an error occurs.
+func TokenIterator[V any](
+	fn func(token *string) (value V, nextToken *string, err error),
+) iter.Seq2[V, error] {
+	return func(yield func(V, error) bool) {
+		var prev *string
+		for {
+			val, next, err := fn(prev)
+			if err != nil {
+				var zero V
+				yield(zero, err)
+				return
+			}
+
+			if !yield(val, nil) {
+				return
+			}
+
+			if next == nil {
+				return
+			}
+			prev = next
+		}
+	}
+}
+
+// Collect takes an iterator of slices and returns a concatentation of all results.
+func Collect[V any](seq iter.Seq2[[]V, error]) ([]V, error) {
+	var result []V
```

Comment:
```suggestion
	result := []V{}
```

This fixes the test "[TestCollectEmptyPages](https://github.com/gregfurman/bento/pull/2/files#diff-0e1c59791a4772848ae03b4f736b50cb94421237c12ed4c5bbdd787bbd71bf3bR65-R70)" 

ie. If there is nothing returned shouldn't we expect an empty slice rather than a nil value?